uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -1394,19 +1394,135 @@ sub make_nii {
+    $nifti     =~ s/mnc$/nii/g;
+    $bval_file =~ s/mnc$/bval/g;
+    $bvec_file =~ s/mnc$/bvec/g;
 
     #  mnc2nii command
     my $m2n_cmd = "mnc2nii -nii -quiet " .
                     $data_dir . "/" . $minc . " " .
                     $data_dir . "/" . $nifti;
     system($m2n_cmd);
 
+    # create complementary nifti files for DWI acquisitions
+    my $bval_success = create_dwi_nifti_bval_file($fileref, $data_dir . '/' . $bval_file);
+    my $bvec_success = create_dwi_nifti_bvec_file($fileref, $data_dir . '/' . $bvec_file);
+
     # update mri table (parameter_file table)
     $file->setParameter('check_nii_filename', $nifti);
+    $file->setParameter('check_bval_filename', $bval_file) if $bval_success;
+    $file->setParameter('check_bvec_filename', $bvec_file) if $bvec_success;
+}
+
+
+=pod
+
+=head3 create_dwi_nifti_bval_file($file_ref, $bval_file)
+
+Creates the NIfTI C<.bval> file required for DWI acquisitions based on the
+returned value of C<acquisition:bvalues>.
+
+INPUTS:
+  - $file_ref : file hash ref
+  - $bval_file: path to the C<.bval> file to write into
+
+RETURNS:
+  - undef if no C<acquisition:bvalues> were found (skipping the creation
+    of the C<.bval> file since there is nothing to write into)
+  - 1 after the C<.bval> file was created
+
+=cut
+
+sub create_dwi_nifti_bval_file {
+    my ($file_ref, $bval_file) = @_;
+
+    # grep bvals from the header acquisition:bvalues
+    my $file  = $$file_ref;
+    my $bvals = $file->getParameter('acquisition:bvalues');
+
+    return undef unless $bvals;
+
+    # clean up the bvals string
+    $bvals =~ s/\.\,//g; # remove all '.,' from the string
+    $bvals =~ s/\.$//g;  # remove the last trailing '.' from the string
+
+    # print bvals into bval_file
+    write_to_file($bval_file, $bvals, '>');
+
+    return 1 if (-e $bval_file);
 }
 
+
+=pod
+
+=head3 create_dwi_nifti_bvec_file($file_ref, $bvec_file)
+
+Creates the NIfTI C<.bvec> file required for DWI acquisitions based on the
+returned value of C<acquisition:direction_x>, C<acquisition:direction_y> and
+C<acquisition:direction_z>.
+
+INPUTS:
+  - $file_ref : file hash ref
+  - $bvec_file: path to the C<.bvec> file to write into
+
+RETURNS:
+  - undef if no C<acquisition:direction_x>, C<acquisition:direction_y> and
+    C<acquisition:direction_z> were found (skipping the creation
+    of the C<.bvec> file since there is nothing to write into)
+  - 1 after the C<.bvec> file was created
+
+=cut
+
+sub create_dwi_nifti_bvec_file {
+    my ($file_ref, $bvec_file) = @_;
+
+    # grep bvecs from headers acquisition:direction_x, y and z
+    my $file  = $$file_ref;
+    my @bvecs = (
+        $file->getParameter('acquisition:direction_x'),
+        $file->getParameter('acquisition:direction_y'),
+        $file->getParameter('acquisition:direction_z')
+    );
+
+    return undef unless @bvecs;
+
+    # loop through all bvecs, clean them up and print them into the bvec file
```

Comment:
I think opening the file, writing to it and then closing it on each iteration is not a good idea. I wouldn't even define a function to print to a file

```perl 
s/^\"+|\"$//g for @bvec;
open(OUT, '>', $bvec_file) or die "Cannot write to file $bvec_file: $!\n";
print OUT map { "$_\n" } @bvec;
close(OUT);
```